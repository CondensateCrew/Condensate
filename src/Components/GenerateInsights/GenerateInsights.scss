@import '../../css/variables.scss';
@import '../../css/index.scss';

.insights-div {
  max-width: 34vw;
  height: fit-content;
  display: flex;
  justify-content: space-between;
  align-items: center;
  @extend %one-line-block;
  border-radius: $one-line-border-radius;
  padding: $one-line-padding-space;
  margin: $small-space auto;
  img {
    cursor: pointer;
    width: 1rem;
    opacity: 0.4;
    @extend %hover-transform;
  }
  h3 {
    font-size: 1rem;
    text-align: left;
    -webkit-hyphens: auto;
    -moz-hyphens: auto;
    -ms-hyphens: auto;
    hyphens: auto;
    overflow-wrap: break-word;
    width: 90%;
  }
  box-shadow: $external-shadow;
}
