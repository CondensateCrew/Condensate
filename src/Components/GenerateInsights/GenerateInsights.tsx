import React from 'react';
import Close from '../../assets/close.svg';
import './GenerateInsights.scss';

interface Props {
  text: string,
  responses: string[],
  setResponses: (responses: string[]) => void
  key:number,
  id:number
}

const GenerateInsights:React.FC<Props> = ({ text, responses, setResponses, key, id }) => {

  const removeResponse = (e: React.SyntheticEvent<HTMLElement>):void => {
    let newResponses = responses;
    newResponses.splice(id, 1)
    setResponses([...newResponses])
  }

  return (
    <div className='insights-div' key={key} id={id.toString()}>
      <h3>{text}</h3>
      <img onClick={removeResponse} src={Close} alt='delete icon' />
    </div>
  )
}

export default GenerateInsights;