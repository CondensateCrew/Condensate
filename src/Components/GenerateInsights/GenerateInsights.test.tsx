/**
 * @jest-environment jsdom
 */

import React from 'react';
import GenerateInsights from './GenerateInsights';
import { shallow } from 'enzyme';

describe('GenerateInsights component', () => {
  let wrapper: any;

  const mockSetResponses = jest.fn();
  const mockProps = {
    text: 'Other',
    responses: ['Response', 'Other'],
    setResponses: mockSetResponses,
    id: 1
  }

  beforeEach(() => {
    wrapper = shallow(<GenerateInsights {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setResponses after remove button was clicked', () => {
    wrapper.find('img').simulate('click');
    expect(mockSetResponses).toHaveBeenCalledWith(['Response']);
  });
});
