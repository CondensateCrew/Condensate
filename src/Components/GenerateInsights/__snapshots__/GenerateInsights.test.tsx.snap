// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateInsights component should match the snapshot 1`] = `
<div
  className="insights-div"
  id="1"
  key="1"
>
  <h3>
    Other
  </h3>
  <img
    alt="delete icon"
    onClick={[Function]}
    src="close.svg"
  />
</div>
`;
